### PR TITLE
have implemented privacy settings for profile fragment

### PR DESCRIPTION
**Pull request by olaokoroafor**
Description:
 - App takes into account privacy settings when determining whether to show profile data

### CODE REVIEW

File: FoodFriends/app/src/main/java/com/example/foodfriends/activities/MapActivity.java [hunk 1]
```diff
@@ -32,12 +35,14 @@
 public class MapActivity extends AppCompatActivity implements OnMapReadyCallback{
     private List<RestaurantObservable> restaurantList;
     private UserObservable user;
+    private String TAG = "Map Activity";
```
Comment (nammyfb) [inline FoodFriends/app/src/main/java/com/example/foodfriends/activities/MapActivity.java:38]: TAGS are generally without a space TAG = "MapActivity"

File: FoodFriends/app/src/main/java/com/example/foodfriends/fragments/ProfileFragment.java [hunk 2]
```diff
@@ -94,11 +97,12 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
      * Connects the recycler view of restaurants to the adapter
      * Populates the list of restaurants for the adapter to display
-     * */
+     */
     @Override
     public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
         currentUser.addObserver(this);
+        loggedInUser = new UserObservable(ParseUser.getCurrentUser());
         rvRestaurants = view.findViewById(R.id.rvProfilePosts);
```
Comment (nammyfb) [inline FoodFriends/app/src/main/java/com/example/foodfriends/fragments/ProfileFragment.java:106]: Same comment about moving these to onCreateView

File: FoodFriends/app/src/main/java/com/example/foodfriends/fragments/ProfileFragment.java [hunk 3]
```diff
@@ -107,83 +111,95 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
         ivFindFriends = view.findViewById(R.id.ivFindFriends);
         btnLogOut = view.findViewById(R.id.btnLogOut);
         ivFollow = view.findViewById(R.id.ivFollow);
+        ivLock = view.findViewById(R.id.ivLock);
 
 
         if (currentUser.getObjectId().equals(ParseUser.getCurrentUser().getObjectId())) {
```
Comment (nammyfb) [inline FoodFriends/app/src/main/java/com/example/foodfriends/fragments/ProfileFragment.java:117]: Quite a bit going on in this method. Be good to refactor into separate methods.

File: FoodFriends/app/src/main/java/com/example/foodfriends/fragments/ProfileFragment.java (hunk 2 above)
Comment (olaokoroafor) [reply FoodFriends/app/src/main/java/com/example/foodfriends/fragments/ProfileFragment.java:106]: Done!

**Review by nammyfb — APPROVED** [review]

**Pull request merged**